### PR TITLE
Increase number of replicas and allow its configuration

### PR DESCRIPTION
**Pull request by migmartri**
Closes https://github.com/kubeapps/kubeapps/issues/643

Diff of rendered template before/after the change.

```bash
 multidoc_yamldiff /tmp/before.yaml /tmp/after.yaml 
@@ .. @@
    name: "RELEASE-NAME-kubeapps",
   },
   spec: {
-   replicas: 1,
+   replicas: 2,
    selector: {
     matchLabels: {
      app: "RELEASE-NAME-kubeapps",
@@ .. @@
    name: "RELEASE-NAME-kubeapps-internal-apprepository-controller",
   },
   spec: {
-   replicas: 1,
+   replicas: 2,
    selector: {
     matchLabels: {
      app: "RELEASE-NAME-kubeapps-internal-apprepository-controller",
@@ .. @@
    name: "RELEASE-NAME-kubeapps-internal-chartsvc",
   },
   spec: {
-   replicas: 1,
+   replicas: 2,
    selector: {
     matchLabels: {
      app: "RELEASE-NAME-kubeapps-internal-chartsvc",
@@ .. @@
    name: "RELEASE-NAME-kubeapps-internal-dashboard",
   },
   spec: {
-   replicas: 1,
+   replicas: 2,
    selector: {
     matchLabels: {
      app: "RELEASE-NAME-kubeapps-internal-dashboard",
@@ .. @@
    name: "RELEASE-NAME-kubeapps-internal-tiller-proxy",
   },
   spec: {
-   replicas: 1,
+   replicas: 2,
    selector: {
     matchLabels: {
      app: "RELEASE-NAME-kubeapps-internal-tiller-proxy",
@@ .. @@
  {
   apiVersion: "v1",
   data: {
-   mongodb-root-password: "SmRMOEg4Nzg5TQ==",
+   mongodb-root-password: "ZTZyOThrQld5RQ==",
   },
   kind: "Secret",
   metadata: {
```

After upgrade

```bash
 k get deployment
NAME                                                DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
miguel-kubeapps                                     2         2         2            2           6d
miguel-kubeapps-internal-apprepository-controller   2         2         2            2           6d
miguel-kubeapps-internal-chartsvc                   2         2         2            2           6d
miguel-kubeapps-internal-dashboard                  2         2         2            2           6d
miguel-kubeapps-internal-tiller-proxy               2         2         2            2           6d
miguel-kubeapps-mongodb                             1         1         1            1           2m
```

### CODE REVIEW

**Comment by andresmgot** [comment]
mm, I think we should leave the default `replicaCount` to 1 and document how to increase it. In my opinion we shouldn't assume that the user wants to run it in HA, should we? In any case I am okay with the changes so go ahead with the number you decide.

File: chart/kubeapps/templates/apprepository-deployment.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
-  replicas: 1
+  replicas: {{ .Values.apprepository.replicaCount }}
```
Comment (prydonius) [inline chart/kubeapps/templates/apprepository-deployment.yaml]: I don't think controllers should have multiple replicas, can you look into this?

Comment (andresmgot) [reply chart/kubeapps/templates/apprepository-deployment.yaml]: Good catch, that would duplicate the sync-jobs.

Comment (migmartri) [reply chart/kubeapps/templates/apprepository-deployment.yaml]: Good point, will look into it. I'd like to also know what are the consequences of having duplicated sync jobs.

Comment (migmartri) [reply chart/kubeapps/templates/apprepository-deployment.yaml]: I confirmed that the controller will create duplicated jobs. I looked into other controller examples like cert-manager and etcd-operator and they seem to run by default with 1 replica, so I put it back to 1.

**Pull request merged**